### ISSUE TITLE
Elimination error

### ISSUE THREAD

**Issue by zsvoboda** [lovasoa/linear-solve#1]
Hi,

can you please help me to fix the error in the  example below? 

It seems that the last two numbers of the solution vector are swapped. 

Thanks!

ZD

```
var ls = require('linear-solve');
var matrix = [
    [-1, 1, 0, 0, 0, -1, 0, 0],
//  [ 2, 6, 6, 2, 4,  4, 2, 0 ]
    [0, -1, 1, 0, 0, 0, -1, 0],
//  [ 2, 6, 6, 2, 4,  4, 2, 0 ] - ERROR: -2 != 0
    [0, 0, 1, -1, 0, -1, 0, 0],
//  [ 2, 6, 6, 2, 4,  4, 2, 0 ]
    [-1, 0, 0, 1, 0, 0, -1, 0],
//  [ 2, 6, 6, 2, 4, 4, 2, 0 ]
    [-1, 0, 0, 0, 1, 0, 0, -1],
//  [ 2, 6, 6, 2, 4, 4, 2, 0 ]
    [0, 0, 1, 0, -1, 0, 0, -1],
//  [ 2, 6, 6, 2, 4, 4, 2, 0 ]
    [ 1, 0, 0, 0, 0, 0, 0, 0],
//  [ 2, 6, 6, 2, 4, 4, 2, 0 ]
    [ 0, 0, 1, 0, 0, 0, 0, 0]];
//  [ 2, 6, 6, 2, 4, 4, 2, 0 ]
var result = [0, 0, 0, 0, 0, 0, 2, 6];
var solution = ls.solve(matrix, result);
console.log('solution', solution); // solution [ 2, 6, 6, 2, 4, 4, 2, 0 ]
```

**Comment by lovasoa** [comment]
Hi!
What do you mean? What solution did you expect ?

**Comment by lovasoa** [comment]
Your matrix[ is not invertible](http://www.wolframalpha.com/input/?i=%5B%5B-1,1,0,0,0,-1,0,0%5D,%5B0,-1,1,0,0,0,-1,0%5D,%5B0,0,1,-1,0,-1,0,0%5D,%5B-1,0,0,1,0,0,-1,0%5D,%5B-1,0,0,0,1,0,0,-1%5D,%5B0,0,1,0,-1,0,0,-1%5D,%5B1,0,0,0,0,0,0,0%5D,%5B0,0,1,0,0,0,0,0%5D%5D). Your equation doesn't have a solution.

**Comment by zsvoboda** [comment]
Thanks for your fast answer! I think that [ 2, 6, 6, 2, 4, 4, 0, 2 ] is one of the correct solutions.

See

    [-1, 1, 0, 0, 0,-1, 0, 0],
//  [ 2, 6, 6, 2, 4, 4, 0, 2 ] = 0
    [ 0,-1, 1, 0, 0, 0,-1, 0],
//  [ 2, 6, 6, 2, 4, 4, 0, 2 ] = 0
    [ 0, 0, 1,-1, 0,-1, 0, 0],
//  [ 2, 6, 6, 2, 4, 4, 0, 2 ] = 0
    [-1, 0, 0, 1, 0, 0,-1, 0],
//  [ 2, 6, 6, 2, 4, 4, 0, 2 ] = 0
    [-1, 0, 0, 0, 1, 0, 0,-1],
//  [ 2, 6, 6, 2, 4, 4, 0, 2 ] = 0
    [ 0, 0, 1, 0,-1, 0, 0,-1],
//  [ 2, 6, 6, 2, 4, 4, 0, 2 ] = 0
    [ 1, 0, 0, 0, 0, 0, 0, 0],
//  [ 2, 6, 6, 2, 4, 4, 0, 2 ] = 2
    [ 0, 0, 1, 0, 0, 0, 0, 0]]
//  [ 2, 6, 6, 2, 4, 4, 0, 2 ] = 6

There are actually many more solutions. Your code computes [ 2, 6, 6, 2, 4, 4, 2, 0 ] which is almost correct. Just the last two numbers of the solution record are swapped.

Thanks!!!

ZD

**Comment by lovasoa** [comment]
Ah, got it. This library is an implementation of the Gauss-Jordan algorithm. Its goal is to solve systems that have exactly one solution (where the matrix is invertible). I added code to make the error explicit instead of just ignoring the case.

**Comment by lovasoa** [comment]
I added support for systems with multiple solutions in the latest commit. I also published it on npm, as version 1.2.0.

Now it [just works](https://runkit.com/580f350772ebca0014ce054e/583a1763c35daa00149bd6da).

Thank you for reporting this initially.

**Comment by zsvoboda** [comment]
Great, thanks a lot for your quick fix!!

ZD
